nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/test/java/org/apache/nifi/processors/aws/kinesis/stream/ITConsumeKinesisStreamEndpointOverride.java

```diff
@@ -49,16 +61,20 @@ public class ITConsumeKinesisStreamEndpointOverride extends ITConsumeKinesisStre
     public void setUp() throws InterruptedException {
         System.setProperty(AWS_CBOR_DISABLE_SYSTEM_PROPERTY, "true");
 
-        kinesis = AmazonKinesisClient.builder()
-                .withEndpointConfiguration(kinesisEndpointConfig)
-                .withCredentials(awsCredentialsProvider)
+        kinesis = KinesisClient.builder()
+                .credentialsProvider(awsCredentialsProvider)
+                .endpointOverride(URI.create(LOCAL_STACK_KINESIS_ENDPOINT_OVERRIDE))
+                .httpClient(ApacheHttpClient.create())
+                .region(Region.of(REGION))
                 .build();
 
-        kinesis.createStream(KINESIS_STREAM_NAME, 1);
+        kinesis.createStream(CreateStreamRequest.builder().streamName(KINESIS_STREAM_NAME).shardCount(1).build());
 
-        dynamoDB = AmazonDynamoDBClient.builder()
-                .withEndpointConfiguration(dynamoDBEndpointConfig)
-                .withCredentials(awsCredentialsProvider)
+        dynamoDB = DynamoDbClient.builder()
+                .credentialsProvider(awsCredentialsProvider)
+                .endpointOverride(URI.create(LOCAL_STACK_DYNAMODB_ENDPOINT_OVERRIDE))
+                .region(Region.of(REGION))
+                .httpClient(ApacheHttpClient.create())
                 .build();
 
         waitForKinesisToInitialize();
```

Comment:
`ACCESS_KEY` and `SECRET_KEY` properties are no longer valid for the processor, they need to be set via the `AWSCredentialsProviderControllerService` instead, e.g.

```java
        final AWSCredentialsProviderControllerService credentialsService = new AWSCredentialsProviderControllerService();
        runner.addControllerService("credentials-service", credentialsService);
        runner.setProperty(credentialsService, CredentialPropertyDescriptors.ACCESS_KEY, ACCESS_KEY);
        runner.setProperty(credentialsService, CredentialPropertyDescriptors.SECRET_KEY, SECRET_KEY);
        runner.enableControllerService(credentialsService);
        runner.setProperty(ConsumeKinesisStream.AWS_CREDENTIALS_PROVIDER_SERVICE, "credentials-service");
```